server/tests/api/check-params/video-channel-syncs.ts

```diff
@@ -0,0 +1,392 @@
+import { checkBadCountPagination, checkBadSortPagination, checkBadStartPagination, FIXTURE_URLS } from '@server/tests/shared'
+import { HttpStatusCode, VideoChannelSyncCreate } from '@shared/models'
+import {
+  ChannelSyncsCommand,
+  createSingleServer,
+  makePostBodyRequest,
+  PeerTubeServer,
+  setAccessTokensToServers,
+  setDefaultVideoChannel
+} from '@shared/server-commands'
+import 'mocha'
+
+describe('Test video channel sync API validator', () => {
+  const path = '/api/v1/video-channel-syncs'
+  let server: PeerTubeServer
+  let command: ChannelSyncsCommand
+  let rootChannelId: number
+  let rootChannelSyncId: number
+  const userInfo = {
+    accessToken: '',
+    username: 'user1',
+    id: -1,
+    channelId: -1,
+    syncId: -1,
+    defaultQuota: -1,
+    defaultDailyQuota: -1
+  }
+
+  async function withChannelSyncDisabled<T> (callback: () => T): Promise<T> {
+    try {
+      await server.config.disableChannelSync()
+      return callback()
```

Comment:
prefer `await callback()` because it's sometimes difficult to understand the finally block will be executed after the `return` :)